File: v1/src/main/java/com/api/v1/mappers/car/CarResponseMapper.java
package com.api.v1.mappers.car;

import com.api.v1.domain.car.Car;
import com.api.v1.dtos.car.CarResponseDto;

public record CarResponseMapper(Car car) {

    public CarResponseDto mapToDto() {
        return new CarResponseDto(
                car.getMake(),
                car.getModel(),
                car.getProductionYear(),
                car.getVin(),
                car.getPlateNumber(),
                car.getAddedAt(),
                car.getUpdatedAt()
        );
    }

}

File: v2/src/main/kotlin/com/api/v2/car/CarResponseMapper.kt
package com.api.v2.car

class CarResponseMapper {

    companion object {
        fun mapToDto(car: Car): CarResponseDto {
            return CarResponseDto(
                car.vin,
                car.model,
                car.productionYear,
                car.make,
                car.plateNumber,
                car.createdAt
            )
        }
    }

}